src/math/dct.c

```diff
@@ -32,53 +32,6 @@
  * multiply with the returned matrix.
  * \param[in,out]  dct  In input provide DCT type and size, in output the DCT matrix
  */
-int dct_initialize_16(struct dct_plan_16 *dct)
-{
-	int16_t dct_val;
-	int32_t arg;
-	int32_t cos;
-	int32_t c1;
-	int16_t c2;
-	int16_t nk;
-	int n;
-	int k;
-
-	if (dct->type != DCT_II || dct->ortho != true)
-		return -EINVAL;
-
-	if (dct->num_in < 1 || dct->num_out < 1)
-		return -EINVAL;
-
-	if (dct->num_in > DCT_MATRIX_SIZE_MAX || dct->num_out > DCT_MATRIX_SIZE_MAX)
-		return -EINVAL;
-
-	dct->matrix = mat_matrix_alloc_16b(dct->num_in, dct->num_out, 15);
-	if (!dct->matrix)
-		return -ENOMEM;
-
-	c1 = PI_Q29 / dct->num_in;
-	arg = Q_SHIFT_RND(TWO_Q29 / dct->num_in, 29, 12);
-	c2 = sqrt_int16(arg); /* Q4.12 */
-	for (n = 0; n < dct->num_in; n++) {
-		for (k = 0; k < dct->num_out; k++) {
-			/* Note: Current int16_t nk works up to DCT_MATRIX_SIZE_MAX = 91 */
-			nk = (Q_SHIFT_LEFT(n, 0, 1) + HALF_Q1) * Q_SHIFT_LEFT(k, 0, 1); /*Q14.2 */
-			arg = Q_MULTSR_32X32((int64_t)c1, nk, 29, 2, 24); /* Q8.24 */
-			/* Note: Q8.24 works up to DCT_MATRIX_SIZE_MAX = 42 */
-			arg %= TWO_PI_Q24;
-			cos = cos_fixed_32b(Q_SHIFT_LEFT(arg, 24, 28)); /* Q1.31 */
-			dct_val = Q_MULTSR_32X32((int64_t)cos, c2, 31, 12, 15); /* Q1.15 */
-			if (k == 0)
-				dct_val = Q_MULTSR_32X32((int64_t)dct_val,
-							 ONE_OVER_SQRT_TWO, 15, 31, 15);
-
-			mat_set_scalar_16b(dct->matrix, n, k, dct_val);
-		}
-	}
-
-	return 0;
-}
```

Comment:
would it be possible to modify these functions in place without first adding new versions and then deleting original ones? E.g. in patch 1 modify function prototypes and add wrappers:
```
-void x(int y)
+void x_new(struct mod *p, int y)
 {
...
-    m = rmalloc(...);
+    if (p)
+        m = mod_alloc(p, ...);
+    else
+        m = rmalloc(...);
 ...
 }
+
+void x(int y)
+{
+    return x_new(NULL, y);
+}
```
then move all users one by one to use `x_new()` and then remove the original `x()` and (optionally) rename `x_new()` back to `x()`?

Reply:
I grant you the point obout being easier to review and to be sure the functions are still in essence the same. But I think our git repository can easily take the duplicated code.

This was a result of my conversion method. Making mod api versions of all that I needed for mfcc and then remove the obsolte functions and see what breaks. I really hope you do not put to redo also this part of this PR series. Good thing of course is that by now I know almost by heart all the allocs and frees in all the module code.